src/v/cluster/shard_placement_table.h

```diff
@@ -284,6 +293,8 @@ class shard_placement_table
 
     chunked_hash_map<model::ntp, std::unique_ptr<entry_t>> _ntp2entry;
     model::shard_revision_id _cur_shard_revision{0};
+
+    std::unique_ptr<probe> _probe;
```

Comment:
what's the point of storing it separately, not inline?

Reply:
It is dangerous to move the probe as metrics reporting closures capture `this` (see also https://github.com/redpanda-data/redpanda/commit/bedfce577767fcd724b0268488a00bf714cb4976). Doesn't really matter much in this case, as `shard_placement_table` is not moved itself, but for consistency I wrapped it in a unique_ptr.

Can probably make it just a member if you want.

Reply:
> Doesn't really matter much in this case, as `shard_placement_table` is not moved itself

Yeah, and the new explicit dtor even guards `shard_placement_table` from moves.

> Can probably make it just a member if you want.

This will need to move the class definition to the header, so no need for explicit dtor, but it's fine because `probe` itself is not movable... IDK I'm probably okay with the PR as it is.